xdsl/dialects/llvm.py

```diff
@@ -361,10 +363,23 @@ def verify(self):
             raise VerifyException(f"Specified linkage '{self.linkage.data}' is unknown")
 
 
+LLVMScalarOrVectorOfInteger = AnyOf(
+    [AnySignlessIntegerType, VectorType[AnySignlessIntegerType]]
+)
+LLVMScalarOrVectorOfFloat = AnyOf([AnyFloat, VectorType[AnyFloat]])
+LLVMScalarOrVectorOfNumeric = AnyOf(
+    [
+        AnySignlessIntegerType,
+        AnyFloat,
+        VectorType[AnySignlessIntegerType | AnyFloat],
+    ]
+)
+
+
 class ArithmeticBinOperation(IRDLOperation, ABC):
     """Class for arithmetic binary operations."""
 
-    T: ClassVar = VarConstraint("T", BaseAttr(IntegerType))
+    T: ClassVar = VarConstraint("T", LLVMScalarOrVectorOfInteger)
```

Comment:
Can you please change this and the ones below? I'd be surprised if it doesn't work, and will be a little neater IMO
```suggestion
    T: ClassVar = VarConstraint("T", AnySignlessIntegerType | VectorType[AnySignlessIntegerType])
```